assets/js/atomic/components/product/button/index.js

```diff
@@ -106,6 +106,15 @@ const ProductButton = ( { product, className } ) => {
 		}
 		return productCartDetails.text;
 	};
+
+	// This is a hack to trigger cart updates till we migrate to block based card
+	// that relies on the store, see
+	// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1247
+	useEffect( () => {
+		const event = document.createEvent( 'Event' );
+		event.initEvent( 'wc_fragment_refresh', true, true );
```

Comment:
[`initEvent`](https://developer.mozilla.org/en-US/docs/Web/API/Event/initEvent) seems to be deprecated. I didn't test it, but do you think we could use `dispatchEvent` instead as shown here:

https://developer.mozilla.org/en-US/docs/Web/API/Event/Event#Example

Reply:
You're right, this is much better